File: tests/fixtures/vulnerable_repo/php/vuln.php
<?php
// Intentionally vulnerable PHP fixture (for Suzaku Compass tests).
// DO NOT use this code. It exists only as scanner bait.

function unsafe_eval($code) {
    eval($code);
}

function unsafe_unserialize($data) {
    $obj = unserialize($data);
    return $obj;
}

function unsafe_include() {
    include $_GET['template'];
}

function unsafe_exec($user_cmd) {
    system($user_cmd);
    shell_exec("ls " . $user_cmd);
}

function unsafe_extract() {
    extract($_POST);
}

// preg_replace with /e modifier (deprecated, but still found in old code)
function unsafe_preg($input) {
    return preg_replace('/foo(.*)/e', 'strtoupper($1)', $input);
}

File: tests/fixtures/vulnerable_repo/java/Vuln.java
// Intentionally vulnerable Java fixture for Suzaku Compass tests.
// DO NOT use this code. It exists only as scanner bait.

import java.io.ObjectInputStream;
import java.io.FileInputStream;
import javax.naming.Context;
import javax.xml.parsers.DocumentBuilderFactory;
import javax.xml.parsers.DocumentBuilder;

public class Vuln {
    public static void unsafeDeserialize(String path) throws Exception {
        ObjectInputStream ois = new ObjectInputStream(new FileInputStream(path));
        Object o = ois.readObject();
    }

    public static void unsafeJndi(Context ctx, String name) throws Exception {
        ctx.lookup(name);
    }

    public static void unsafeExec(String cmd) throws Exception {
        Runtime.getRuntime().exec(cmd);
    }

    public static void unsafeXml(String xml) throws Exception {
        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
        DocumentBuilder builder = factory.newDocumentBuilder();
    }
}
